particle-iot/particle-cli
# feature/sc-126221/upload-artifacts-to-s3-from-ci

## Description
This PR will:
* Generate CLIInstaller for windows (offline installation)
* Add the ability to publish new binaries to an s3 bucket
* Generate (When pushing a new version) a manifest.json for the current version being published and also (in case is not a pre-release) update the current "latest" manifest.json

> [!NOTE]
> All tests and publish jobs were disabled (commented) in the yaml.
> This should be restored in next PR's




## How to Test
* Visit the latest circleCI execution
* Check a circleCI execution with a tag `test-3.23.3`
* Check [manifest.json](https://binaries.staging.particle.io/particle-cli/manifest.json)
* Attempt to download the urls for each binary in the manifest


**Outcome** 
* In CircleCI artifacts for that execution you will see all binaries and the cli Windows Installer
* The tag will have a `publish` step 
* You will be able to see the manifest and download the files by using the urls into the document



## Related Issues / Discussions
Epic: https://app.shortcut.com/particle/epic/126209/cli-installer-v2
Story details: 
* [Upload to S3](https://app.shortcut.com/particle/story/126221/upload-artifacts-to-s3-from-ci)
* [Add Cli Installer](https://app.shortcut.com/particle/story/126215/include-cli-executable-in-windows-installer)




## Completeness
